codegen/type_converter.go

```diff
@@ -297,9 +320,10 @@ func (c *TypeConverter) genConverterForList(
 			fromFieldType.ValueSpec,
 			valID,
 			keyPrefix+pascalCase(toField.Name)+"["+indexID+"]",
-			strings.TrimPrefix(fromIdentifier, "in.")+"["+indexID+"]",
+			strings.TrimPrefix(strings.TrimPrefix(fromIdentifier, "in."), "inOriginal.")+"["+indexID+"]",
```

Comment:
nit: `inOriginal` and `outOriginal` are constant declared upfront.

Reply:
up

Reply:
I have to declare them as constants upfront, otherwise later inside some function literal I won't be able to access the original ```in``` and ```out```. Normally I wouldn't need to but fieldMap overrides are written in terms of the original input (hence inOriginal). outOriginal is needed for nil-checking when a fieldMap has an assignment to an optional to-field with no corresponding from-field

Reply:
maybe not nit, I think variable scope is a bit out of control as @oliviazhang0809 suggested, if you need to access some variables set miles away from where it used, it's usually a good sign to refactor :)

I'll leave this one to @oliviazhang0809

Reply:
I agree it is quite out of control.. I'm open to alternate solutions, but the dilemma here is to have recursive sub-methods (only way to handle recursive structs?), which are naturally local to a type, but also support fieldMap, whose mapping could be from anywhere. How else would this local sub-method have access to any arbitrary variable?

Reply:
what i meant is to have 
```
const (
	inOriginal = "inOriginal"
	outOriginal = "outOriginal"
)
```
upfront -- global scope and use the variable in functions. - genernally best practise.

Reply:
Oh I see. Though Go is not letting me write
```
const (
    inOriginal = in
)
```
because ```in``` is not a constant